### PR TITLE
musescore: Add wrapGAppsHook3

### PR DESCRIPTION
**Pull request by niklasmohrin**
## Description of changes

When using musescore, I get the error

> No GSettings schemas are installed on the system

when trying to save a file on my computer (the program crashes right after selecting "File > Save As > Save to computer"). I tried installing `gsettings-desktop-schemas` and `musescore` in my NixOS configuration and with home-manager, as well as running musescore with `nix run` on a flake with nixpkgs-unstable and nixos-24.05 directly. I found https://github.com/NixOS/nixpkgs/issues/181500 and https://github.com/NixOS/nixpkgs/issues/263383 and tried the fix for the latteer from https://github.com/NixOS/nixpkgs/pull/285912 and it worked. The changes here are exactly the same as from this PR and, as far as I can tell, match the instructions on https://ryantm.github.io/nixpkgs/languages-frameworks/gnome/#ssec-gnome-common-issues-double-wrapped.

Please let me know if there is something I missed! :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/audio/musescore/default.nix [hunk 1]
```diff
@@ -95,7 +96,14 @@ in stdenv'.mkDerivation (finalAttrs: {
...
   ];
 
+  preFixup = ''
+    qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  dontWrapGApps = true;
+
   nativeBuildInputs = [
+    wrapGAppsHook
```
Comment (Aleksanaa) [inline pkgs/applications/audio/musescore/default.nix:106]:
```suggestion
    wrapGAppsNoGuiHook
```

Comment (niklasmohrin) [reply pkgs/applications/audio/musescore/default.nix:106]: I am not sure I understand this change, the problem occurs exactly when a GUI is supposed to be displayed, but the program crashes. In fact, I tested with this change applied and it reverted to the broken behavior.

Can you explain why the no-gui version is expected to work so we can figure out what went wrong?

Comment (Aleksanaa) [reply pkgs/applications/audio/musescore/default.nix]: The difference between `wrapGAppsHook3` and `wrapGAppsNoGuiHook` is just the former propagates `gtk3` and `librsvg`, which should not be needed as you just want some environment variables to find schemas. See https://github.com/NixOS/nixpkgs/blob/36316edca39a1f1575613d5ca71bffade3e8a814/pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix#L20

**Review by Aleksanaa — APPROVED** [review]

**Comment by niklasmohrin** [comment]
@Aleksanaa So now you merged it without the no-gui change, so is it correct like this?

**Comment by Aleksanaa** [comment]
It's not accurate but it works

**Comment by doronbehar** [comment]
My perfectionism wouldn't let me leave this as is, so: https://github.com/NixOS/nixpkgs/pull/342089 . Thanks @Aleksanaa for explaining the subtlety.

**Pull request merged**